fetch.js

```diff
@@ -186,7 +186,9 @@
     }
 
     this.json = function() {
-      return this.text().then(JSON.parse)
+      return this.text().then(function(text) {
+        return JSON.parse(text)
+      })
```

Comment:
Yeah it seems like it'd be better to report this Firefox bug on their bugtracker.